src/polus/plugins/_plugins/manifests/manifest_utils.py

```diff
@@ -66,35 +70,41 @@ def _load_manifest(m: typing.Union[str, dict, pathlib.Path]) -> dict:
         else:
             manifest = requests.get(m).json()
     else:
-        raise ValueError("invalid manifest")
+        msg = "invalid manifest"
+        raise ValueError(msg)
     return manifest
 
 
 def validate_manifest(
-    manifest: typing.Union[str, dict, pathlib.Path]
+    manifest: typing.Union[str, dict, pathlib.Path],
 ) -> typing.Union[WIPPPluginManifest, ComputeSchema]:
     """Validate a plugin manifest against schema."""
     manifest = _load_manifest(manifest)
     manifest["version"] = cast_version(
-        manifest["version"]
+        manifest["version"],
     )  # cast version to semver object
+    if "name" in manifest:
+        name = manifest["name"]
+    else:
+        raise InvalidManifest(f"{manifest} has no value for name")
+
     if "pluginHardwareRequirements" in manifest:
         # Parse the manifest
         try:
             plugin = ComputeSchema(**manifest)
-        except ValidationError as err:
-            raise err
+        except ValidationError as e:
+            raise InvalidManifest(f"{name} does not conform to schema") from e
         except BaseException as e:
             raise e
     else:
         # Parse the manifest
         try:
             plugin = WIPPPluginManifest(**manifest)
-        except ValidationError as err:
-            logger.info(manifest)
-            raise err
+        except ValidationError as e:
+            raise InvalidManifest(
+                f"{manifest['name']} does not conform to schema"
+            ) from e
         except BaseException as e:
-            logger.info(manifest)
             raise e
```

Comment:
You could use `logger.critical` in such cases. In this case though, I don't think that's a good idea. The manifests can get really large.